sigopt/request_driver.py

```diff
@@ -81,18 +120,24 @@ def _request(self, method, url, params, json, headers, user_agent):
       raise ConnectionException('\n'.join(message)) from rqe
     return response
 
-  def request(self, method, url, params=None, json=None, headers=None, user_agent=None):
+  def request(self, method, path, data, headers):
+    method = method.upper()
+    url = "/".join(str(v) for v in (self.api_url, self.api_version, *path))
+    if method in ('GET', 'DELETE'):
+      json, params = None, self._request_params(data)
+    else:
+      json, params = ApiObject.as_json(data), None
     retry = backoff.on_predicate(
       backoff.expo,
       lambda response: response.status_code == HTTPStatus.TOO_MANY_REQUESTS,
       max_time=self.timeout,
       jitter=backoff.full_jitter,
     )
-    response = retry(self._request)(method, url, params, json, headers, user_agent)
+    response = retry(self._request)(method, url, params, json, headers)
     return self._handle_response(response)
 
-  def _with_default_headers(self, headers, user_agent):
-    user_agent_str = user_agent or 'sigopt-python/{0}'.format(VERSION)
+  def _with_default_headers(self, headers):
+    user_agent_str = 'sigopt-python/{0}'.format(VERSION)
```

Comment:
tidbit - no longer python2, so we can probably just `f"sigopt-python/{VERSION}"`?